### PR TITLE
pep8 fixes, remove egg directory, doc update

### PR DESCRIPTION
**Pull request by tomkralidis**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ before_install:
     - source activate test
     - travis_retry conda install --yes pytest
     - conda info -a
+    - pip install -r requirements-dev.txt
```
Comment (ocefpaf) [inline .travis.yml]: @tomkralidis can you "fold" the `pytest` into `requirements-dev.txt`, remove the `pytest` line above and install `requirements-dev.txt` wth conda?

File: .travis.yml [hunk 2]
```diff
@@ -20,3 +21,4 @@ install:
 
 script:
     - cd yaml2ncml/tests && py.test
+    - find . -type f -name "*.py"|xargs flake8 --max-line-length=100
```
Comment (ocefpaf) [inline .travis.yml]: :+1:

File: .travis.yml (hunk 1 above)
Comment (tomkralidis) [inline .travis.yml]: Fixed/rebased/pushed.

**Comment by ocefpaf** [comment]
Rich, I am  :+1: to this PR, but I do not have the rights to merge it.

**Comment by rsignell-usgs** [comment]
Thanks Tom!

**Pull request merged**